src/Components/test/E2ETest/ServerExecutionTests/WebSocketCompressionTests.cs

```diff
@@ -50,6 +50,39 @@ public async Task EmbeddingServerAppInsideIframe_WorksAsync()
             Assert.DoesNotContain("Content-Security-Policy", response.Headers.Select(h => h.Key));
         }
     }
+
+    [Fact]
+    public async Task EmbeddingServerAppInsideIframe_WorksWithMultipleCspHeaders()
+    {
+        Navigate("/subdir/iframe?add-csp");
+
+        var logs = Browser.GetBrowserLogs(LogLevel.Severe);
+
+        Assert.Empty(logs);
+
+        // Get the iframe element from the page, and inspect its contents for a p element with id inside-iframe
+        var iframe = Browser.FindElement(By.TagName("iframe"));
+        Browser.SwitchTo().Frame(iframe);
+        Browser.Exists(By.Id("inside-iframe"));
+
+        using var client = new HttpClient() { BaseAddress = _serverFixture.RootUri };
+        var response = await client.GetAsync("/subdir/iframe?add-csp");
+        response.EnsureSuccessStatusCode();
+
+        if (ExpectedPolicy != null)
+        {
+            Assert.Equal(
+                response.Headers.GetValues("Content-Security-Policy"),
+                [
+                    "script-src 'self' 'unsafe-inline'",
+                    $"frame-ancestors {ExpectedPolicy}"
```

Comment:
Nice. I was imagining that we might need to merge content together in the header content, but since browsers support multiple CSP headers (today I learned) this is much cleaner.

Reply:
yes, they "and" all the policies, so you can't relax it, but you can strengthen it.